pkg/rebuild/maven/gradle.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2025 Google LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package maven
+
+import (
+	"encoding/json"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/pkg/errors"
+)
+
+const gradleScript = `import groovy.json.JsonOutput
+
+gradle.projectsLoaded {
+    def outputJsonPath = gradle.startParameter.projectProperties['outputJson']
+
+    gradle.rootProject.tasks.register("printCoordinates") {
+        doLast {
+            def jsonFile = new File(outputJsonPath ?: "output.json")
+
+            def rootProjectPath = gradle.rootProject.getProjectDir().toPath()
+
+            def toJsonMap
+            toJsonMap = { Project project ->
+                def map = [
+                        groupId   : project.getGroup().toString(),
+                        artifactId: project.getName(),
+                        version   : project.getVersion().toString(),
+                        buildManifest: rootProjectPath.relativize(project.getBuildFile().toPath()).toString(),
+                ]
+                if (!project.getChildProjects().isEmpty()) {
+                    map.submodules = project.getChildProjects().values().collect {
+                        toJsonMap(it)
+                    }
+                }
+                return map
+            }
+
+            def jsonModel = toJsonMap(gradle.getRootProject())
+            def jsonString = JsonOutput.prettyPrint(JsonOutput.toJson(jsonModel))
+            jsonFile.text = jsonString
+        }
+    }
+}`
```

Comment:
@loosebazooka need your feedback here. Specifically, line 30, 31, 32, and 33 where I fetch GAV from [Project](https://docs.gradle.org/9.0.0/javadoc/org/gradle/api/Project.html). You said something about immutability of these properties; the values of these properties may change after some point in the build phase. Do you think this approach is a good way to get the GAV since it runs before the build starts? Would we either miss (return empty values) or infer wrong GAV for some cases? We simply run `./gradlew printCoordinates --init-script printCoordinates.gradle`.